cpp/mg_utility/mg_utils.hpp

```diff
@@ -143,75 +168,55 @@ std::unique_ptr<mg_graph::Graph<TSize>> GetGraphView(mgp_graph *memgraph_graph,
+}
+
 template <typename TSize = std::uint64_t>
 std::unique_ptr<mg_graph::Graph<TSize>> GetWeightedGraphView(mgp_graph *memgraph_graph, mgp_result *result,
                                                              mgp_memory *memory, const mg_graph::GraphType graph_type,
                                                              const char *weight_property, double default_weight) {
-  auto graph = std::make_unique<mg_graph::Graph<TSize>>();
-
-  ///
-  /// Mapping Memgraph in-memory vertices into graph view
-  ///
-
-  // Safe vertices iterator creation
-  auto *vertices_it = mgp::graph_iter_vertices(memgraph_graph, memory);
-  mg_utility::OnScopeExit delete_vertices_it([&vertices_it] { mgp::vertices_iterator_destroy(vertices_it); });
-
-  // Iterate through Memgraph vertices and map them to GraphView
-  for (auto *vertex = mgp::vertices_iterator_get(vertices_it); vertex;
-       vertex = mgp::vertices_iterator_next(vertices_it)) {
-    mg_graph::CreateGraphNode(graph.get(), vertex);
-  }
-  // Destroy iterator before creating a new one in order to avoid memory leakage
-  mgp::vertices_iterator_destroy(vertices_it);
-
-  ///
-  /// Mapping Memgraph in-memory edges into graph view
-  ///
-
-  // Safe vertices iterator creation
-  vertices_it = mgp::graph_iter_vertices(memgraph_graph, memory);
-
-  for (auto *vertex_from = mgp::vertices_iterator_get(vertices_it); vertex_from;
-       vertex_from = mgp::vertices_iterator_next(vertices_it)) {
-    // Safe edges iterator creation
-    auto *edges_it = mgp::vertex_iter_out_edges(vertex_from, memory);
-    mg_utility::OnScopeExit delete_edges_it([&edges_it] { mgp::edges_iterator_destroy(edges_it); });
-
-    for (auto *out_edge = mgp::edges_iterator_get(edges_it); out_edge; out_edge = mgp::edges_iterator_next(edges_it)) {
-      auto vertex_to = mgp::edge_get_to(out_edge);
-      auto weight = mg_utility::GetNumericProperty(out_edge, weight_property, memory, default_weight);
-      mg_graph::CreateWeightedGraphEdge(graph.get(), vertex_from, vertex_to, weight, graph_type);
-    }
-  }
+  return GraphViewGetter(memgraph_graph, result, memory, graph_type, true, weight_property, default_weight);
+}
 
-  return graph;
+template <typename TSize = std::uint64_t>
+std::unique_ptr<mg_graph::Graph<TSize>> GetWeightedSubgraphView(mgp_graph *memgraph_graph, mgp_result *result,
+                                                                mgp_memory *memory, mgp_list *subgraph_nodes,
+                                                                mgp_list *subgraph_relationships,
+                                                                const mg_graph::GraphType graph_type,
+                                                                const char *weight_property, double default_weight) {
+  return GraphViewGetter(memgraph_graph, result, memory, graph_type, true, weight_property, default_weight, true,
+                         subgraph_nodes, subgraph_relationships);
 }
```

Comment:
If we want to organize it like this, isn't then `GetSubgraphView` function missing (unweighted + subgraph), since we have
`GetWeightedSubgraphView` (weight + sub),
`GetWeightedGraphView` (weight + graph),
and `GetGraphView` (unweight + graph)?

Reply:
Agreed here :arrow_up: